diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter2/CommentsFormattingTest.kt

```diff
@@ -0,0 +1,382 @@
+                    |   val a = 5// This is a comment
+                    |}
+                """.trimMargin()
+
+        lintMethod(code,
+                LintError(5,13, ruleId, "${Warnings.COMMENT_WHITE_SPACE.warnText()} // This is a comment", true))
+    }
+
+    @Test
+    @Tag(WarningNames.IF_ELSE_COMMENTS)
+    fun `if - else comments good` () {
+        val code =
+                """
+                    |package org.cqfn.diktat.ruleset.chapter3
+                    |
+                    |class Example {
+                    |   fun someFunc() {
+                    |       // general if comment
+                    |       if(a = 5) {
+                    |       
+                    |       }
+                    |       else {
+                    |           // Good Comment
+                    |           print(5)
+                    |       }
+                    |   }
+                    |}
+                """.trimMargin()
+
+        lintMethod(code)
+    }
+
+    @Test
+    @Tag(WarningNames.IF_ELSE_COMMENTS)
+    fun `if - else comments good 2` () {
+        val code =
+                """
+                    |package org.cqfn.diktat.ruleset.chapter3
+                    |
+                    |class Example {
+                    |   fun someFunc() {
+                    |       // general if comment
+                    |       if(a = 5) {
+                    |       
+                    |       } else {
+                    |           // Good Comment
+                    |           print(5)
+                    |       }
+                    |   }
+                    |}
+                """.trimMargin()
+
+        lintMethod(code)
+    }
+
+    @Test
+    @Tag(WarningNames.IF_ELSE_COMMENTS)
+    fun `if - else comments bad` () {
+        val code =
+                """
+                    |package org.cqfn.diktat.ruleset.chapter3
+                    |
+                    |class Example {
+                    |   fun someFunc() {
+                    |       // general if comment
+                    |       if(a = 5) {
+                    |       
+                    |       }
+                    |       // Bad Comment
+                    |       else {
+                    |           print(5)
+                    |       }
+                    |   }
+                    |}
+                """.trimMargin()
+
+        lintMethod(code,
+                LintError(6,8,ruleId, "${IF_ELSE_COMMENTS.warnText()} // Bad Comment", true))
+    }
+
+    @Test
+    @Tag(WarningNames.IF_ELSE_COMMENTS)
+    fun `if - else comments bad 2` () {
+        val code =
+                """
+                    |package org.cqfn.diktat.ruleset.chapter3
+                    |
+                    |class Example {
+                    |   fun someFunc() {
+                    |       // general if comment
+                    |       if(a = 5) {
+                    |       
+                    |       } else { // Bad Comment 
```

Comment:
Why is this considered as an error in this rule? Comment is inside `else`, and newlines are checked in another rule